### ISSUE TITLE
[Question] How do I config the mod?

### ISSUE THREAD

**Issue by tazoi2** [Bstn1802/AutoReconnect#43]
I am very new to Fabric mods. I've been playing 1.8.9 PvP for the longest time but when I finally found a good vanilla SMP, I joined it but it required 1.19.1-1.19.3. After my application was accepted, I installed Fabric (1.19.2 because there are more mods for that version). Another member of the SMP sent this mod in the Discord and I realized that I really needed this to AFK fish farm overnight despite restarts happening at ~5 am every night. But the server reopens after ~2 minutes and because the server tries 4 times in about 1 minute, I can't rejoin the server overnight and I'm pretty sure my fish rates are being affected by this.

TLDR: help config please idk why i typed this entire thing

**Comment by lizquilty** [comment]
I also have no idea how to config this, doesnt work with mod menu, is there a key binding?

**Comment by Bstn1802** [comment]
(Technically you can configure the .json config manually if you know what you're doing, it's in the minecraft directory under config I think), but:
I recommend to use ModMenu, it's another mod which will allow you to configure mods that support ModMenu in game, which this mod does. So if you have both installed you'll see an additional button on the title screen which says... uh, idk rn actually, something with "mods", it should be obvious. Anyways, you click on that and you'll see a list of mods on the left side and when you click on one you get more details to the right including, if the mod supports it, a button to configure it, somewhere to the top right. If you click that you should end up on a screen that looks like the screenshot in the readme.md here on github or on the curseforge page. Hopefully it is selfexplanatory how to configure that or at least with the given instructions, again, in the readme or on the curseforge page, in game you'll see some tooltips as well. If it's unclear, you can ask here and maybe I can even improve on the instructions or in game look.
Aditionally you can get to the mod menu screen from in game too, from the menu that opens when you hit escape, there should be a button somewhere close to the exit button, one or two buttons above it. So you don't have to exit the game.
Simple way to test btw is to disconnect from your internet until the game disconnects you, it should try reconnect a couple times depending on how you configured it.

**Comment by lizquilty** [comment]
Mod menu isnt working, thats the issue. Also i did have a look around and see no .json file in the config  - though its loaded fine and seems to be reconnecting, id love to tweak it some

**Comment by Bstn1802** [comment]
Thats weird, is it not even showing up in the list or is it just not configurable?
There should also definitely be a `autoreconnect.json` in the config folder. Is there a folder called `fabric` inside too or any othere `.json` files? Otherwise it might not be the correct folder. If it is, you might want to send some log files then for me to inspect, maybe I can find an error in there. For that you best just start the game and exit again and then copy the `.../logs/latest.log` from your minecraft directory, the one where fabric is installed, might be a different one depending on how you've set it up.

**Comment by lizquilty** [comment]
It just wasnt showing up at all in the gui anywhere in mod menu. There is a fabric folder and other json files in the config dir, but none for the autoreconnect .
Looking at the logs i see

```
[12:51:09] [Render thread/WARN]: AutoReconnect could not load the config
java.nio.file.NoSuchFileException: /home/liz/.minecraft/config/autoreconnect.json
        at sun.nio.fs.UnixException.translateToIOException(UnixException.java:92) ~[?:?]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:106) ~[?:?]
        at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111) ~[?:?]
        at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:218) ~[?:?]
        at java.nio.file.Files.newByteChannel(Files.java:380) ~[?:?]
        at java.nio.file.Files.newByteChannel(Files.java:432) ~[?:?]
        at java.nio.file.Files.readAllBytes(Files.java:3288) ~[?:?]
        at java.nio.file.Files.readString(Files.java:3366) ~[?:?]
        at java.nio.file.Files.readString(Files.java:3325) ~[?:?]
        at autoreconnect.config.AutoReconnectConfig.load(AutoReconnectConfig.java:39) ~[autoreconnect-2.1.0-beta.jar:?]
        at autoreconnect.AutoReconnect.onInitializeClient(AutoReconnect.java:34) ~[autoreconnect-2.1.0-beta.jar:?]
        at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke0(EntrypointUtils.java:47) [fabric-loader-0.14.9.jar:?]
        at net.fabricmc.loader.impl.entrypoint.EntrypointUtils.invoke(EntrypointUtils.java:35) [fabric-loader-0.14.9.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.Hooks.startClient(Hooks.java:53) [fabric-loader-0.14.9.jar:?]
        at net.minecraft.class_310.<init>(class_310.java:459) [client-intermediary.jar:?]
        at net.minecraft.client.main.Main.method_44604(Main.java:205) [client-intermediary.jar:?]
        at net.minecraft.client.main.Main.main(Main.java:51) [client-intermediary.jar:?]
        at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:461) [fabric-loader-0.14.9.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) [fabric-loader-0.14.9.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.9.jar:?]
:

```

**Comment by lizquilty** [comment]
Just found this also
```
[12:51:34] [Render thread/ERROR]: Error from mod 'autoreconnect'
java.lang.NoSuchMethodError: 'net.minecraft.class_4185$class_7840 net.minecraft.class_4185.method_46430(net.minecraft.class_2561, net.minecraft.class_4185$class_4241)'
        at me.shedaniel.clothconfig2.gui.entries.BaseListEntry.<init>(BaseListEntry.java:95) ~[cloth-config-10.0.95-fabric.jar:?]
        at me.shedaniel.clothconfig2.gui.entries.AbstractListListEntry.<init>(AbstractListListEntry.java:53) ~[cloth-config-10.0.95-fabric.jar:?]
        at me.shedaniel.clothconfig2.gui.entries.AbstractTextFieldListListEntry.<init>(AbstractTextFieldListListEntry.java:55) ~[cloth-config-10.0.95-fabric.jar:?]
        at me.shedaniel.clothconfig2.gui.entries.IntegerListListEntry.<init>(IntegerListListEntry.java:54) ~[cloth-config-10.0.95-fabric.jar:?]
        at me.shedaniel.clothconfig2.impl.builders.IntListBuilder.build(IntListBuilder.java:152) ~[cloth-config-10.0.95-fabric.jar:?]
        at autoreconnect.config.ModMenuIntegration.createConfigScreen(ModMenuIntegration.java:39) ~[autoreconnect-2.1.0-beta.jar:?]
        at com.terraformersmc.modmenu.ModMenu.getConfigScreen(ModMenu.java:52) ~[modmenu-4.0.6.jar:?]
        at com.terraformersmc.modmenu.gui.ModsScreen.method_25426(ModsScreen.java:117) ~[modmenu-4.0.6.jar:?]
        at net.minecraft.class_437.method_41843(class_437.java:389) ~[client-intermediary.jar:?]
        at net.minecraft.class_437.method_25423(class_437.java:380) ~[client-intermediary.jar:?]
        at net.minecraft.class_310.method_1507(class_310.java:1035) ~[client-intermediary.jar:?]
        at com.terraformersmc.modmenu.gui.widget.ModMenuButtonWidget.lambda$new$0(ModMenuButtonWidget.java:11) ~[modmenu-4.0.6.jar:?]
        at net.minecraft.class_4185.method_25306(class_4185.java:43) ~[client-intermediary.jar:?]
        at net.minecraft.class_4264.method_25348(class_4264.java:16) ~[client-intermediary.jar:?]
        at net.minecraft.class_339.method_25402(class_339.java:120) ~[client-intermediary.jar:?]
        at net.minecraft.class_4069.method_25402(class_4069.java:27) ~[client-intermediary.jar:?]
        at net.minecraft.class_442.method_25402(class_442.java:514) ~[client-intermediary.jar:?]
        at net.minecraft.class_312.method_1611(class_312.java:94) ~[client-intermediary.jar:?]
        at net.minecraft.class_437.method_25412(class_437.java:489) ~[client-intermediary.jar:?]
        at net.minecraft.class_312.method_1601(class_312.java:94) ~[client-intermediary.jar:?]
        at net.minecraft.class_312.method_22686(class_312.java:165) ~[client-intermediary.jar:?]
        at net.minecraft.class_1255.execute(class_1255.java:118) ~[client-intermediary.jar:?]

```

**Comment by lizquilty** [comment]
Actually looking at it, getting the same error from cloth config, and autofish

**Comment by Bstn1802** [comment]
The first error just says that the file doesn't exists, which as far as I remember is intentional, it catches the error and then creates the default config which should be saved... at some point I guess? If its not saved it doesnt matter, it should use the default config everytime then when it sees there is no config file. As soon as you change the config in game, it should definitely save the config, but seeing that there's another error with cloth config, I assume that you need another version of that. At least I assume that error has to do with cloth config, although it says it's from autoreconnect but it sure looks like it has something to do with clothconfig. Yeah so check your cloth config version again and maybe change to another version, let me know if that fixed it then.

**Comment by tezoon** [comment]
I have the same problem, you can send config file? maybe we could just insert it...
